File: 11. Container With Most Water/11.py
"""
When searching for a pair in a 2D list, consider eliminating
multiple pairs at one time
"""
def maxArea(self, heights):
  l = 0
  r = len(heights) - 1
  result = 0;
  while(l < r):
    if(heights[l] < heights[r]):
      result = max(result, (r - l) * heights[l])
      l += 1
    else:
      result = max(result, (r - l) * heights[r])
      r -= 1
  return result

File: 11. Container With Most Water/11.java
/**
 * When searching for a pair in a 2D list, consider eliminating
 * multiple pairs at one time
 */
public int maxArea(int[] height) {
  int l = 0, r = height.length - 1;
  int result = 0;
  while(l < r) {
    if(height[l] < height[r]) {
      result = Math.max(result, (r - l) * height[l++]);
    }
    else {
      result = Math.max(result, (r - l) * height[r--]);
    }
  }
  return result;
}